neuralmonkey/learning_utils.py

```diff
@@ -280,8 +280,18 @@ def run_on_dataset(tf_manager: TensorFlowManager,
     if postprocess is not None:
         for series_name, postprocessor in postprocess:
             postprocessed = postprocessor(dataset, result_data)
+            if not hasattr(postprocessed, '__len__'):
+                postprocessed = list(postprocessed)
+
             result_data[series_name] = postprocessed
 
+    # check output series lengths
+    for series_id, data in result_data.items():
+        if len(data) != len(dataset):
+            warn("Output '{}' for dataset '{}' has length {}, but "
```

Comment:
Ono to je asi fuk, protože si ten striktní režim stejně nikdo nezapne mimo ty testy...